xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.cpp

```diff
@@ -255,6 +270,122 @@ ComPtr<ID3D11VideoProcessorInputView> CProcessorHD::GetInputView(CRenderBuffer*
+  {
+    ApplyFilter(D3D11_VIDEO_PROCESSOR_FILTER_CONTRAST, static_cast<int>(contrast), 0, 100, 50);
+
+    m_lastContrast = contrast;
+    updatedParameter = true;
+  }
+
+  // unused filters - set once and forget
+  if (m_firstRender)
+  {
+    ApplyFilter(D3D11_VIDEO_PROCESSOR_FILTER_HUE, 50, 0, 100, 50);
+    ApplyFilter(D3D11_VIDEO_PROCESSOR_FILTER_SATURATION, 50, 0, 100, 50);
+  }
+
+  if (m_firstRender || m_lastRotation != rotation)
+  {
+    m_pVideoContext->VideoProcessorSetStreamRotation(
+        m_pVideoProcessor.Get(), DEFAULT_STREAM_INDEX, rotation != 0,
+        static_cast<D3D11_VIDEO_PROCESSOR_ROTATION>(rotation / 90));
+
+    m_lastRotation = rotation;
+    updatedParameter = true;
+  }
+
+  ComPtr<ID3D11VideoContext1> videoCtx1;
+  if (SUCCEEDED(m_pVideoContext.As(&videoCtx1)))
+  {
+    if (m_firstRender || m_lastConversion != m_conversion)
+    {
+      videoCtx1->VideoProcessorSetStreamColorSpace1(m_pVideoProcessor.Get(), DEFAULT_STREAM_INDEX,
+                                                    m_conversion.m_inputCS);
+      videoCtx1->VideoProcessorSetOutputColorSpace1(m_pVideoProcessor.Get(),
+                                                    m_conversion.m_outputCS);
+
+      m_lastConversion = m_conversion;
+      updatedParameter = true;
+    }
+  }
+  else if (m_firstRender || m_lastColorSpace != rb.color_space || m_lastFullRange != rb.full_range)
+  {
+    // input colorspace
+    bool isBT601 = rb.color_space == AVCOL_SPC_BT470BG || rb.color_space == AVCOL_SPC_SMPTE170M;
+    // clang-format off
+    D3D11_VIDEO_PROCESSOR_COLOR_SPACE colorSpace
+    {
+      0u,                             // 0 - Playback, 1 - Processing
+      rb.full_range ? 0u : 1u, // 0 - Full (0-255), 1 - Limited (16-235) (RGB)
+      isBT601 ? 1u : 0u,              // 0 - BT.601, 1 - BT.709
+      0u,                             // 0 - Conventional YCbCr, 1 - xvYCC
+      rb.full_range ? 2u : 1u  // 0 - driver defaults, 2 - Full range [0-255], 1 - Studio range [16-235] (YUV)
+    };
+    // clang-format on
+    m_pVideoContext->VideoProcessorSetStreamColorSpace(m_pVideoProcessor.Get(),
+                                                       DEFAULT_STREAM_INDEX, &colorSpace);
+    // Output color space
+    // don't apply any color range conversion, this will be fixed at later stage.
+    colorSpace.Usage = 0; // 0 - playback, 1 - video processing
+    colorSpace.RGB_Range = DX::Windowing()->UseLimitedColor() ? 1 : 0; // 0 - 0-255, 1 - 16-235
+    colorSpace.YCbCr_Matrix = 1; // 0 - BT.601, 1 = BT.709
+    colorSpace.YCbCr_xvYCC = 1; // 0 - Conventional YCbCr, 1 - xvYCC
+    colorSpace.Nominal_Range = 0; // 2 - 0-255, 1 = 16-235, 0 - undefined
+    m_pVideoContext->VideoProcessorSetOutputColorSpace(m_pVideoProcessor.Get(), &colorSpace);
+
+    m_lastColorSpace = rb.color_space;
+    m_lastFullRange = rb.full_range;
+    updatedParameter = true;
+  }
+
+  if (m_firstRender)
+    m_firstRender = false;
```

Comment:
if you change to `m_configured` method then is more intuitive:

```c++
if (updatedParameter)
  m_configured = true;
```

`m_configured` is already true nothing bad happens and anyway this code is not executed at every frame.

Reply:
I don't follow how that's more intuitive. m_configured is a guard to detect the first time Render() and CheckVideoParameters() are called. That way there is no need to rely on special initialization values of m_last* (not as robust).

The parameters must be set once for first Render() even in worst case and every parameter is equal to first value passed to Render.

It could be without condition:
```m_configured = true```
or
```
if (!m_configured)
  m_configured = true;
```
Doesn't make much difference for a single variable.

It is called for every frame but the dxva processor functions are not.

Changed the members tracking last processed frame to use the same mechanism instead of relying on UINT_MAX.